Source/Controllers/Preferences/Panes/SPNetworkPreferencePane.m

```diff
@@ -520,30 +520,65 @@ + (NSArray *)defaultSSLCipherList
 {
 	//this is the default list as hardcoded in SPMySQLConnection.m
 	//Sadly there is no way to make MySQL give us the list of runtime-supported ciphers.
-	return @[@"DHE-RSA-AES256-SHA",
-			 @"AES256-SHA",
-			 @"DHE-RSA-AES128-SHA",
-			 @"AES128-SHA",
-			 @"AES256-RMD",
-			 @"AES128-RMD",
-			 @"DES-CBC3-RMD",
-			 @"DHE-RSA-AES256-RMD",
-			 @"DHE-RSA-AES128-RMD",
-			 @"DHE-RSA-DES-CBC3-RMD",
-			 @"RC4-SHA",
-			 @"RC4-MD5",
-			 @"DES-CBC3-SHA",
-			 @"DES-CBC-SHA",
-			 @"EDH-RSA-DES-CBC3-SHA",
-			 @"EDH-RSA-DES-CBC-SHA",
-			 SPSSLCipherListMarkerItem, //marker. disabled items below here
-			 @"EDH-DSS-DES-CBC-SHA",
-			 @"EDH-DSS-DES-CBC3-SHA",
-			 @"DHE-DSS-AES128-SHA",
-			 @"DHE-DSS-AES256-SHA",
-			 @"DHE-DSS-DES-CBC3-RMD",
-			 @"DHE-DSS-AES128-RMD",
-			 @"DHE-DSS-AES256-RMD"];
+	return @[
+		@"ECDHE-ECDSA-AES128-GCM-SHA256",
+		@"ECDHE-ECDSA-AES256-GCM-SHA384",
+		@"ECDHE-RSA-AES128-GCM-SHA256",
+		@"ECDHE-ECDSA-AES128-SHA256",
+		@"ECDHE-RSA-AES128-SHA256",
+		@"ECDHE-RSA-AES256-GCM-SHA384",
+		@"ECDHE-ECDSA-AES256-SHA384",
+		@"ECDHE-RSA-AES256-SHA384",
+		@"DHE-RSA-AES128-GCM-SHA256",
+		@"DHE-DSS-AES128-GCM-SHA256",
+		@"DHE-RSA-AES128-SHA256",
+		@"DHE-DSS-AES128-SHA256",
+		@"DHE-DSS-AES256-GCM-SHA384",
+		@"DHE-RSA-AES256-SHA256",
+		@"DHE-DSS-AES256-SHA256",
+		@"DHE-RSA-AES256-GCM-SHA384",
+		@"DH-DSS-AES128-GCM-SHA256",
+		@"ECDH-ECDSA-AES128-GCM-SHA256",
+		@"DH-DSS-AES256-GCM-SHA384",
+		@"ECDH-ECDSA-AES256-GCM-SHA384",
+		@"DH-DSS-AES128-SHA256",
+		@"ECDH-ECDSA-AES128-SHA256",
+		@"DH-DSS-AES256-SHA256",
+		@"ECDH-ECDSA-AES256-SHA384",
+		@"DH-RSA-AES128-GCM-SHA256",
+		@"ECDH-RSA-AES128-GCM-SHA256",
+		@"DH-RSA-AES256-GCM-SHA384",
+		@"ECDH-RSA-AES256-GCM-SHA384",
+		@"DH-RSA-AES128-SHA256",
+		@"ECDH-RSA-AES128-SHA256",
+		@"DH-RSA-AES256-SHA256",
+		@"ECDH-RSA-AES256-SHA384",
+		@"ECDHE-RSA-AES128-SHA",
+		@"ECDHE-ECDSA-AES128-SHA",
+		@"ECDHE-RSA-AES256-SHA",
+		@"ECDHE-ECDSA-AES256-SHA",
+		@"DHE-RSA-AES128-SHA",
+		@"DHE-RSA-AES256-SHA",
+		@"DH-DSS-AES128-SHA",
+		@"ECDH-ECDSA-AES128-SHA",
+		@"AES256-SHA",
+		@"DH-DSS-AES256-SHA",
+		@"ECDH-ECDSA-AES256-SHA",
+		@"DH-RSA-AES128-SHA",
+		@"ECDH-RSA-AES128-SHA",
+		@"DH-RSA-AES256-SHA",
+		@"ECDH-RSA-AES256-SHA",
+		@"CAMELLIA256-SHA",
+		@"CAMELLIA128-SHA",
+		@"AES128-GCM-SHA256",
+		@"AES256-GCM-SHA384",
+		@"AES128-SHA256",
+		@"AES256-SHA256",
+		@"AES128-SHA",
+		SPSSLCipherListMarkerItem, //marker. disabled items below here
+		@"DHE-DSS-AES128-SHA",
```

Comment:
Can we just comment this out? I don't know how the marker works, but seems "safer" to not include them at all

Reply:
Hmmm we could comment them out, but as long as the marker works as intended (which testing in #164 indicates it does) there's no inherent danger in including less secure ciphers as long as they are disabled by default. I worry that if we remove these less secure ciphers, some users won't be able to connect to their legacy databases which may not support the more secure ciphers at all. This way, they would have to go in and manually allow a less secure cipher, but we're not making the app unusable for these users.